src/test/java/uk/gov/pay/api/it/CreateAgreementIT.java

```diff
@@ -66,12 +70,87 @@ public void shouldReturn400WhenWhenReferenceIsNull() {
                 .body("description", is("Missing mandatory attribute: reference"));
     }
 
+    @Test
+    public void shouldReturn400WhenWhenDescriptionIsNull() {
+        publicAuthMockClient.mapBearerTokenToAccountId(API_KEY, GATEWAY_ACCOUNT_ID, CARD);
+        CreateAgreementRequest agreementRequest = new CreateAgreementRequest(CreateAgreementRequestBuilder.builder().reference(REFERENCE).description(null));
+        postAgreementRequest(agreementRequest.toConnectorPayload())
+                .statusCode(HttpStatus.SC_BAD_REQUEST)
+                .contentType(JSON)
+                .body("field", is("description"))
+                .body("code", is("P0101"))
+                .body("description", is("Missing mandatory attribute: description"));
+    }
+
+    @Test
+    public void shouldReturn400WhenWhenDescriptionIsEmptyString() {
+        publicAuthMockClient.mapBearerTokenToAccountId(API_KEY, GATEWAY_ACCOUNT_ID, CARD);
+        CreateAgreementRequest agreementRequest = new CreateAgreementRequest(CreateAgreementRequestBuilder.builder().reference(REFERENCE).description(""));
+        postAgreementRequest(agreementRequest.toConnectorPayload())
+                .statusCode(HttpStatus.SC_BAD_REQUEST)
+                .contentType(JSON)
+                .body("field", is("description"))
+                .body("code", is("P0101"))
+                .body("description", is("Missing mandatory attribute: description"));
+    }
+
+    @Test
+    public void shouldReturn422WhenWhenDescriptionIsTooLong() {
+        publicAuthMockClient.mapBearerTokenToAccountId(API_KEY, GATEWAY_ACCOUNT_ID, CARD);
+        CreateAgreementRequest agreementRequest = new CreateAgreementRequest(CreateAgreementRequestBuilder.builder().reference(REFERENCE).description(random(256, true, true)));
+        postAgreementRequest(agreementRequest.toConnectorPayload())
+                .statusCode(HttpStatus.SC_UNPROCESSABLE_ENTITY)
+                .contentType(JSON)
+                .body("field", is("description"))
+                .body("code", is("P0102"))
+                .body("description", is("Invalid attribute value: description. Must be less than or equal to 255 characters length"));
+    }
+
+    @Test
+    public void shouldReturn201WhenWhenOptionalUserIdentifierIsNull() {
```

Comment:
I assume this behaviour is consistent with other optional fields in our API.

Reply:
Yes